check-for-unicode/run.sh

```
@@ -127,6 +131,91 @@ is_common_unicode() {
+        # Databases
+        db|sqlite|sqlite3)
+            return 0
+            ;;
+        # Other binary formats
+        bin|dat|pkg|deb|rpm|dmg)
+            return 0
+            ;;
+    esac
+    
+    # Use file command to check MIME type
+    local mime_type=$(file -b --mime-type "$file" 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)